src/sentry/integrations/github/client.py

```diff
@@ -509,7 +506,7 @@ def get_repositories(self, fetch_max_pages: bool = False) -> Sequence[Any]:
         repos = self.get_with_pagination(
             "/installation/repositories",
             response_key="repositories",
-            page_number_limit=self.page_number_limit if fetch_max_pages else 1,
+            page_number_limit=self.page_number_limit,
```

Comment:
This will make a maximum of 50 pages. Sometimes, this may not be all repositories but that's okay.